src/main/java/ortus/boxlang/runtime/cli/BoxRepl.java

```diff
@@ -0,0 +1,517 @@
+			char c = input.charAt( i );
+
+			// Handle escape characters in strings
+			if ( escapeChar != 0 ) {
+				escapeChar = 0;
+				continue;
+			}
+
+			// Check for escape sequences
+			if ( ( inString || inSingleQuote ) && c == '\\' ) {
+				escapeChar = c;
+				continue;
+			}
+
+			// Handle comments
+			if ( !inString && !inSingleQuote ) {
+				// Start of line comment
+				if ( !inBlockComment && i < input.length() - 1 && c == '/' && input.charAt( i + 1 ) == '/' ) {
+					inLineComment = true;
+					i++; // Skip the second '/'
+					continue;
+				}
+				// Start of block comment
+				if ( !inLineComment && i < input.length() - 1 && c == '/' && input.charAt( i + 1 ) == '*' ) {
+					inBlockComment = true;
+					i++; // Skip the '*'
+					continue;
+				}
+				// End of block comment
+				if ( inBlockComment && i < input.length() - 1 && c == '*' && input.charAt( i + 1 ) == '/' ) {
+					inBlockComment = false;
+					i++; // Skip the '/'
+					continue;
+				}
+			}
+
+			// Skip characters inside comments
+			if ( inLineComment || inBlockComment ) {
+				continue;
+			}
+
+			// Handle string boundaries
+			if ( !inSingleQuote && c == '"' ) {
+				inString = !inString;
+				continue;
+			}
+			if ( !inString && c == '\'' ) {
+				inSingleQuote = !inSingleQuote;
+				continue;
+			}
+
+			// Count braces only when not inside strings or comments
+			if ( !inString && !inSingleQuote ) {
+				if ( c == '{' ) {
+					depth++;
+				} else if ( c == '}' ) {
+					depth--;
+				}
+			}
+		}
+
+		return depth;
+	}
+
+	/**
+	 * BoxLang-specific syntax highlighter implementation
+	 */
+	private class BoxLangSyntaxHighlighter implements ISyntaxHighlighter {
+
+		private static final Pattern	COMPONENT_PATTERN	= Pattern.compile( "\\bbx:([a-zA-Z][a-zA-Z0-9_]*)\\b" );
+		private static final Pattern	BIF_PATTERN			= Pattern.compile( "\\b([a-zA-Z][a-zA-Z0-9_]*)\\s*\\(" );
+		private Set<String>				bifs				= new HashSet<>();
+		private Set<String>				components			= new HashSet<>();
+
+		/**
+		 * Constructor uses the shared BIFs and components from the outer REPL class.
+		 */
+		public BoxLangSyntaxHighlighter() {
+			// Use the outer class's shared bifs and components sets
+			this.bifs		= BoxRepl.this.bifs;
+			this.components	= BoxRepl.this.components;
+		}
+
+		@Override
+		public String highlight( String input ) {
+			if ( input == null || input.isEmpty() ) {
+				return input;
+			}
+			String			workingInput		= input;
+
+			// First, highlight component declarations (bx:componentName)
+			Matcher			componentMatcher	= COMPONENT_PATTERN.matcher( workingInput );
+			StringBuffer	tempBuffer			= new StringBuffer();
+
+			while ( componentMatcher.find() ) {
+				String componentName = componentMatcher.group( 1 ).toLowerCase();
+				if ( components.contains( componentName ) ) {
+					// Highlight the entire bx:componentName in cyan
+					String highlighted = MiniConsole.color( 51 ) + componentMatcher.group( 0 ) + MiniConsole.reset();
```

Comment:
My one piece of feedback on this is that the colors are hard-coded throughout.  This is going to favor dark mode and could make some things hard to read if the terminal background is a light color.   

I might suggest abstracting these in to a color palette so you can choose the correct colors, depending on the mode or background.